### PR TITLE
Broken co-author link - (OCT-295) - BUG

### PR DESCRIPTION
**Pull request by shaunajisc**
The purpose of this PR was to fix a broken co author link on the 'view a publication' page. The link was using the user's ORCHID id in the path params for the URL, instead of the `linkedUser` id. I have now amended it so it uses the `linkedUser` id.

### CODE REVIEW

File: ui/src/components/Publication/Creation/DataStatements.tsx [hunk 1]
```diff
@@ -43,7 +43,15 @@ const DataStatements: React.FC = (): React.ReactElement => {
     // Data access statement
     const dataAccessStatement = Stores.usePublicationCreationStore((state) => state.dataAccessStatement);
     const updateDataAccessStatement = Stores.usePublicationCreationStore((state) => state.updateDataAccessStatement);
-    const [dataAccessStatementOther, setDataAccessStatementOther] = React.useState('');
+    const [dataAccessStatementOther, setDataAccessStatementOther] = React.useState<string | null>('');
```
Comment (karmseveeratjisc) [inline ui/src/components/Publication/Creation/DataStatements.tsx:46]: this seems to revert a change elsewhere and also is unrelated to the fix

File: ui/src/lib/types.ts [hunk 2]
```diff
@@ -71,7 +71,7 @@ export type PublicationCreationStoreType = {
     ethicalStatementFreeText: string | null;
     updateEthicalStatementFreeText: (ethicalStatementFreeText: string | null) => void;
     updateEthicalStatement: (ethicalStatement: string) => void;
-    dataAccessStatement: string | null;
+    dataAccessStatement: string;
```
Comment (karmseveeratjisc) [inline ui/src/lib/types.ts:74]: again this is not related to the fix (and is already merged, I find github a bit strange in that respect bitbucket would show this as either not a change or a conflict)

**Comment by CarlyRich** [comment]
@karmseveeratjisc If you want to re-review please :) thanks!

**Review by karmseveeratjisc — APPROVED** [review]
